tests/bsim/compile.source

```diff
@@ -37,6 +38,11 @@ function _compile(){
     local exe_basename="${exe_name:-bs_${BOARD}_${app}_${conf_file}}"
   fi
 
+  if [ "${file_suffix}" ]; then
+    cmake_args+=" -DFILE_SUFFIX='${file_suffix}'"
+    exe_basename+="_FILE_SUFFIX_${file_suffix}"
```

Comment:
you could override the exe name from the caller instead and add any extra cmake arguments from there also if you really need this

Reply:
I just want to clarify that `FILE_SUFFIX` is a [Zephyr build system variable](https://docs.zephyrproject.org/latest/develop/application/index.html#important-build-system-variables), just like `CONF_FILE`. It's not something ad-hoc for this sample.